spectree/plugins/falcon_plugin.py

```diff
@@ -206,12 +211,31 @@ def response_validation(
                 if all(isinstance(entry, expected_list_item_type) for entry in model):
                     skip_validation = True
                 falcon_response.media = [
-                    (entry.dict() if isinstance(entry, BaseModel) else entry)
+                    (
+                        serialize_model_instance(entry)
+                        if isinstance(entry, BaseModel)
+                        else entry
+                    )
                     for entry in model
                 ]
+            elif (
+                expect_model
+                and is_root_model(expect_model)
+                and not isinstance(model, expect_model)
+            ):
+                # Make it possible to return an instance of the model __root__ type
+                # (i.e. not the root model itself).
+                try:
+                    model = expect_model(__root__=model)
+                except ValidationError:
+                    raise
```

Comment:
This is okay to raise here in the falcon plugin, because it catches these errors upstream